### ISSUE TITLE
Warp spacing issues

### ISSUE THREAD

**Issue by rainrdx** [xpipe-io/xpipe#104]
Thank you again for working on this.

The app works quite well with Warp under mac, but there is a very weird spacing issue.

Every time I open up with Warp as terminal, there is a massive blank under the prompt. and every time I type, warp jumps the focus to the bottom of the screen, which is the blank space, causing it very difficult to type up any command. Please see a screen shot as follows:

![image](https://github.com/xpipe-io/xpipe/assets/6933788/3e9eae00-18da-41be-9452-13a0e244eb1b)

Anytime I type up anything, it jumps to the bottom of the screen:

![image](https://github.com/xpipe-io/xpipe/assets/6933788/34ff6d18-dbd5-4382-9777-32a5e5fc93cc)

**Comment by crschnick** [comment]
I assume that you use the classic input position? In Warp settings -> Appearance -> Input.

I suspect that the clear screen command used by xpipe in init files is the cause: https://github.com/xpipe-io/xpipe/blob/b146b9bb275f7ab77e818865dd01fa09bb82b372/app/src/main/java/io/xpipe/app/util/ScriptHelper.java#L57

Does manually clearing this screen with `printf "\e[1;1H\e[2J"` cause the same issue?

**Comment by rainrdx** [comment]
Wow thank you.

I use pin to the bottom and yes manually clearing the screen causes the same issue.

**Comment by rainrdx** [comment]
I dont wanna add too many issues and not sure this is user error or not. Under warp or terminal.app., it does seem the command shortcuts don't work, including tab for autocomplete and up for history of commands?

**Comment by crschnick** [comment]
For the screen clearing, I think I can just use the `clear` command instead. it seems like Warp has specific overrides for it which makes it work there as well.

The command shortcuts should work, that must be another bug. I just tried that, they work for me

**Comment by rainrdx** [comment]
Thank you. I tried the shortcuts on terminal.app. does not work either. Thank you!

**Comment by crschnick** [comment]
Is there anything special about your environment? Because xpipe is not doing a lot that should influence that.

**Comment by rainrdx** [comment]
Not really. Honestly I intentionally keep my mac very very clean and simple and I don't even I even installed a lot of apps. I restarted XPipe and it works now. very weird..I would ignore that part for now. Thank you already for debugging with me!

**Comment by crschnick** [comment]
So the clear issue should be now fixed in the latest version of https://github.com/xpipe-io/xpipe-ptb.

If you somehow managed to reproduce the shortcuts not working, then I can take a look at it.

**Comment by rainrdx** [comment]
Thank you I will give it a try today or tomorrow. Thank you!

**Comment by rainrdx** [comment]
@crschnick I can confirm all of the problems have been solved. The only last question I have is that the use of SSH keys in the data folder isn't quite clear. I don't know how to refer to the file in the connection setting. No doubt it works if I choose identity file and then put in the absolute path. But because I am traveling now, I am not sure if upon syncing to Windows, the absolute path still converts. I can also use ~/.xpipe-ptb/storage/data/XXX.key also worries if the relative path or even the .xpipe-ptb part converts in a different computer/OS or between ptb and formal releases. Thank you maybe that is just user error. Everything else works perfectly.

**Comment by crschnick** [comment]
I reworked the clear screen again because I still had some issues (did you switch back to the old warp display mode?) in https://github.com/xpipe-io/xpipe-ptb/releases/tag/1.7.9-3. That also includes more git fixes plus all the paths should convert properly in the data directory

**Comment by rainrdx** [comment]
Thank you enormously for spending time to work on those. I'm trying it out tonight.

**Comment by crschnick** [comment]
1. Whoops, that will be fixed
2. That should also be fixable
3. That was a recent introduction for another issue, but will also be fixed
4. From you screenshot, I think that this relates to `sh` not having that feature. Any other shell like bash or zsh should work. That should not be xpipe specific, if you manually run `sh` you should encounter the same issue

**Comment by rainrdx** [comment]
> 1. Whoops, that will be fixed
> 2. That should also be fixable
> 3. That was a recent introduction for another issue, but will also be fixed
> 4. From you screenshot, I think that this relates to `sh` not having that feature. Any other shell like bash or zsh should work. That should not be xpipe specific, if you manually run `sh` you should encounter the same issue

Ah thank you. I didnt notice yes this is sh. so yes 4 is a non issue.

Thank you!

**Comment by rainrdx** [comment]
Just one more thing:

when I was using 1.7.8 with broken git, it would ask me there was a conflict with repo, and if we I want to override local, override upstream, or nothing. But ever since 1.7.9, I never get such a conflict notice. Even when I created an empty readme.md on upstream, it seems that local just force overwrite upstream without even asking me (which in practice is fine, but probably should have been a confirmation).

**Comment by crschnick** [comment]
Alright so the https://github.com/xpipe-io/xpipe-ptb/releases/tag/1.7.9-4 should include fixes for everything (hopefully)

Note that get repositories from older versions might cause initial merge conflict. If you create a new repository with the latest version and share it, there shouldn't be any merge conflicts on initial pull.

**Comment by rainrdx** [comment]
ahh sorry. Seems one more issue. I created a new repository and it is empty without any commits. this is the error message:

fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

**Comment by crschnick** [comment]
Did you create the repository manually or did xpipe do that automatically?

**Comment by rainrdx** [comment]
> Did you create the repository manually or did xpipe do that automatically?

Interesting! I always created it manually. I don't even know that xpipe does it. might just be my own user error.

**Comment by crschnick** [comment]
If you enable the git storage, it will call git init on the next restart

**Comment by rainrdx** [comment]
> If you enable the git storage, it will call git init on the next restart

Ohhh. I misunderstood. For every new PTB, I always remove the entire ./xpipe-ptb folder, enable git storage to let it automatically init, and create a new repo on gitbhub.

But I figured out the problem:

If I enable the git storage, restart, and THEN enable the remote git address. Everything works perfectly.

Previously, I always just enabled the git storage AND remote git address at the same time. And then when I restart the app, I saw errors.
